Userland/Libraries/LibGUI/EmojiInputDialog.cpp

```diff
@@ -247,13 +253,31 @@ void EmojiInputDialog::update_displayed_emoji()
                     found_match = result.score > 0;
                 }
 
-                if (found_match)
+                if (found_match) {
                     horizontal_container.add_child(*emoji.button);
+                    m_displayed_emojis.append(emoji);
```

Comment:
Storing a `Vector` of all displayed emoji seems overly expensive when we only want the first one. Instead, let's just store a pointer to the first emoji we come across.

So on the class definition, you can add `Emoji const* m_first_displayed_emoji { nullptr };`.
Then at the beginning of `EmojiInputDialog::update_displayed_emoji`, reset it to `nullptr`.
Then here, do:

```c++
if (found_match) {
    horizontal_container.add_child(*emoji.button);

    if (m_first_displayed_emoji == nullptr)
        m_first_displayed_emoji = &emoji;
}
````

Reply:
Thank you for your suggestions. I created a new commit with the requested changes. I hope everything checks out. :smile: